### ISSUE TITLE
Can not display existing tasks after installation

### ISSUE THREAD

**Issue by yangli1-stanford** [theunraveler/taskwarrior-web#69]
No task appears after installation using sudo gem install taskwarrior-web.
I can use the "Add task" button to create a new task. The new task can be viewed using taskwarrior's command line interface, but is not displayed on the web interface.

I'm using taskwarror-web-1.1.11, versions are
rack 1.2
sinatra 1.4.5
vegas 0.1.11

taskwarrior version is 2.4.0.beta3

OS is Ubuntu 14.10

**Comment by patschwork** [comment]
Same with me... 
It worked with taskwarrior 2.0.0.0. I've update to stable 2.4 an since then there is the problem already mentioned above.

**Comment by growdigital** [comment]
Same issue here
taskwarrior 2.4.1
rack 1.3
sinatra 1.4.4
vegas 0.1.11
Mac OS X 10.9.5

**Comment by growdigital** [comment]
Contents of `$ task show`
```
Config Variable                 Value
_forcecolor                     no
abbreviation.minimum            2
active.indicator                *
alias.burndown                  burndown.weekly
alias.ghistory                  ghistory.monthly
alias.history                   history.monthly
alias.rm                        delete
alias.shell                     exec tasksh
allow.empty.filter              yes
avoidlastcolumn                 no
bulk                            3
burndown.bias                   0.666
calendar.details                sparse
calendar.details.report         list
calendar.holidays               none
calendar.legend                 yes
calendar.offset                 no
calendar.offset.value           -1
color                           on
color.active                    black on bright green
color.alternate
color.blocked                   black on white
color.blocking                  black on bright white
color.burndown.done             on green
color.burndown.pending          on red
color.burndown.started          on yellow
color.calendar.due              white on red
color.calendar.due.today        bold white on red
color.calendar.holiday          black on bright yellow
color.calendar.overdue          black on bright red
color.calendar.today            bold white on bright blue
color.calendar.weekend          white on bright black
color.calendar.weeknumber       bold blue
color.completed                 green on white
color.debug                     yellow
color.deleted                   red on white
color.due                       red
color.due.today                 red
color.error                     yellow
color.footnote                  yellow
color.header                    yellow
color.history.add               black on red
color.history.delete            black on yellow
color.history.done              black on green
color.label
color.label.sort
color.overdue                   bold red
color.pri.H                     bold white
color.pri.L
color.pri.M                     white
color.pri.none
color.project.none
color.recurring                 magenta
color.scheduled                 on green
color.summary.background        white on black
color.summary.bar               black on green
color.sync.added                green
color.sync.changed              yellow
color.sync.rejected             red
color.tag.next                  bold yellow
color.tag.none
color.tagged                    green
color.undo.after                green
color.undo.before               red
color.until
color.warning                   bold red
column.padding                  1
complete.all.tags               no
confirmation                    yes
data.location                   ~/.task
date.iso                        yes
dateformat                      Y-M-D
dateformat.annotation
dateformat.edit                 Y-M-D H:N:S
dateformat.holiday              YMD
dateformat.info                 Y-M-D H:N:S
dateformat.report
debug                           no
default.command                 next
default.due
default.priority
default.project
defaultheight                   24
defaultwidth                    80
dependency.confirmation         on
dependency.indicator            D
dependency.reminder             on
detection                       on
displayweeknumber               yes
dom                             on
due                             7
exit.on.missing.db              no
expressions                     infix
fontunderline                   yes
gc                              on
hooks                           on
hyphenate                       on
indent.annotation               2
indent.report                   0
journal.info                    on
journal.time                    no
journal.time.start.annotation   Started task
journal.time.stop.annotation    Stopped task
json.array                      off
list.all.projects               no
list.all.tags                   no
locking                         on
nag                             You have more urgent tasks.
print.empty.columns             no
recurrence                      yes
recurrence.confirmation         prompt
recurrence.indicator            R
recurrence.limit                1
regex                           yes
report.active.columns           id,start,start.age,entry.age,depends.indicator,priority,project,tags,recur,wait.indicator,scheduled.remaining,due,until,description
report.active.description       Active tasks
report.active.filter            status:pending and +ACTIVE
report.active.labels            ID,Started,Active,Age,D,P,Project,Tags,Recur,W,Sch,Due,Until,Description
report.active.sort              project+,start+
report.all.columns              id,status.short,uuid.short,start.active,entry.age,end.age,depends.indicator,priority,project.parent,tags.count,recur.indicator,wait.remaining,scheduled.remaining,due,until.remaining,description
report.all.description          All tasks
report.all.labels               ID,St,UUID,A,Age,Done,D,P,Project,Tags,R,Wait,Sch,Due,Until,Description
report.all.sort                 entry-
report.blocked.columns          id,depends,project,priority,due,start.active,entry.age,description
report.blocked.description      Blocked tasks
report.blocked.filter           status:pending +BLOCKED
report.blocked.labels           ID,Deps,Proj,Pri,Due,Active,Age,Description
report.blocked.sort             due+,priority-,start-,project+
report.blocking.columns         id,uuid.short,start.active,depends,project,tags,recur,wait.indicator,scheduled.remaining,due.remaining,until.remaining,description.count,urgency
report.blocking.description     Blocking tasks
report.blocking.filter          status:pending +BLOCKING
report.blocking.labels          ID,UUID,A,Deps,Project,Tags,R,W,Sch,Due,Until,Description,Urg
report.blocking.sort            urgency-,due+,entry+
report.completed.columns        id,uuid.short,entry,end,entry.age,depends,priority,project,tags,recur.indicator,due,description
report.completed.description    Completed tasks
report.completed.filter         status:completed
report.completed.labels         ID,UUID,Created,Completed,took,Deps,P,Project,Tags,R,Due,Description
report.completed.sort           end+
report.list.columns             id,start.age,entry.age,depends.indicator,priority,project,tags,recur.indicator,scheduled.countdown,due,until.remaining,description.count,urgency
report.list.description         Most details of tasks
report.list.filter              status:pending
report.list.labels              ID,Active,Age,D,P,Project,Tags,R,Sch,Due,Until,Description,Urg
report.list.sort                start-,due+,project+/,urgency-
report.long.columns             id,start.active,entry,modified.age,depends,priority,project,tags,recur,wait.remaining,scheduled,due,until,description
report.long.description         All details of tasks
report.long.filter              status:pending
report.long.labels              ID,A,Created,Mod,Deps,P,Project,Tags,Recur,Wait,Sched,Due,Until,Description
report.long.sort                modified-
report.ls.columns               id,start.active,depends.indicator,project,tags,recur.indicator,wait.remaining,scheduled.countdown,due.countdown,until.countdown,description.count
report.ls.description           Few details of tasks
report.ls.filter                status:pending
report.ls.labels                ID,A,D,Project,Tags,R,Wait,S,Due,Until,Description
report.ls.sort                  start-,description+
report.minimal.columns          id,project,tags.count,description.count
report.minimal.description      Minimal details of tasks
report.minimal.filter           (status:pending or status:waiting)
report.minimal.labels           ID,Project,Tags,Description
report.minimal.sort             project+/,description+
report.newest.columns           id,start.age,entry,entry.age,modified.age,depends.indicator,priority,project,tags,recur.indicator,wait.remaining,scheduled.countdown,due,until.age,description
report.newest.description       Newest tasks
report.newest.filter            (status:pending or status:waiting) and recur.none:
report.newest.labels            ID,Active,Created,Age,Mod,D,P,Project,Tags,R,Wait,Sch,Due,Until,Description
report.newest.sort              entry-
report.next.columns             id,start.age,entry.age,depends,priority,project,tags,recur,scheduled.countdown,due.remaining,until.remaining,description,urgency
report.next.description         Most urgent tasks
report.next.filter              status:pending limit:page
report.next.labels              ID,Active,Age,Deps,P,Project,Tag,Recur,S,Due,Until,Description,Urg
report.next.sort                start-,urgency-
report.oldest.columns           id,start.age,entry,entry.age,modified.age,depends.indicator,priority,project,tags,recur.indicator,wait.remaining,scheduled.countdown,due,until.age,description
report.oldest.description       Oldest tasks
report.oldest.filter            (status:pending or status:waiting)
report.oldest.labels            ID,Active,Created,Age,Mod,D,P,Project,Tags,R,Wait,Sch,Due,Until,Description
report.oldest.sort              entry+
report.overdue.columns          id,start.age,entry.age,depends,priority,project,tags,recur.indicator,scheduled.countdown,due,until,description,urgency
report.overdue.description      Overdue tasks
report.overdue.filter           (status:pending or status:waiting) and +OVERDUE
report.overdue.labels           ID,Active,Age,Deps,P,Project,Tag,R,S,Due,Until,Description,Urg
report.overdue.sort             urgency-,due+
report.ready.columns            id,start.age,entry.age,depends.indicator,priority,project,tags,recur.indicator,scheduled.countdown,due.countdown,until.remaining,description,urgency
report.ready.description        Most urgent actionable tasks
report.ready.filter             +READY
report.ready.labels             ID,Active,Age,D,P,Project,Tags,R,S,Due,Until,Description,Urg
report.ready.sort               start-,urgency-
report.recurring.columns        id,start.age,entry.age,depends.indicator,priority,project,tags,recur,scheduled.countdown,due,until.remaining,description,urgency
report.recurring.description    Recurring Tasks
report.recurring.filter         (status:pending or status:waiting) and (+PARENT or +CHILD)
report.recurring.labels         ID,Active,Age,D,P,Project,Tags,Recur,Sch,Due,Until,Description,Urg
report.recurring.sort           due+,urgency-,entry+
report.unblocked.columns        id,depends,project,priority,due,start.active,entry.age,description
report.unblocked.description    Unblocked tasks
report.unblocked.filter         status:pending -BLOCKED
report.unblocked.labels         ID,Deps,Proj,Pri,Due,Active,Age,Description
report.unblocked.sort           due+,priority-,start-,project+
report.waiting.columns          id,start.active,entry.age,depends.indicator,priority,project,tags,recur.indicator,wait,wait.remaining,scheduled,due,until,description
report.waiting.description      Waiting (hidden) tasks
report.waiting.filter           +WAITING
report.waiting.labels           ID,A,Age,D,P,Project,Tags,R,Wait,for,Sched,Due,Until,Description
report.waiting.sort             due+,wait+,entry+
reserved.lines                  1
row.padding                     0
rule.precedence.color           deleted,completed,active,keyword.,tag.,uda.,project.,overdue,scheduled,due.today,due,blocked,blocking,recurring,tagged,pri.
search.case.sensitive           yes
shell.prompt                    task>
tag.indicator                   +
taskd.ciphers                   NORMAL
taskd.trust                     strict
undo.style                      side
urgency.active.coefficient      4.0
urgency.age.coefficient         2.0
urgency.age.max                 365
urgency.annotations.coefficient 1.0
urgency.blocked.coefficient     -5.0
urgency.blocking.coefficient    8.0
urgency.due.coefficient         12.0
urgency.inherit.coefficient     0.0
urgency.next.coefficient        15.0
urgency.priority.coefficient    6.0
urgency.project.coefficient     1.0
urgency.scheduled.coefficient   5.0
urgency.tags.coefficient        1.0
urgency.waiting.coefficient     -3.0
verbose                         yes
weekstart                       sunday
xterm.title                     no
```

**Comment by ray65536** [comment]
Same issue, 0 tasks shown, but  I can see them through tw cli, and add via web.
TW 2.4.1
[Task show](https://gist.github.com/ray65536/ae36a88322b61943778a)
ruby 2.1.5p273 (2014-11-13 revision 48405) [x86_64-linux]   \(Also tried 1.9.3)
gem 2.4.6

**Comment by sanjayankur31** [comment]
Confirming. Nothing shows up in task-web. Any ideas how we could get more info to help debug this issue?

**Comment by Tylerlhess** [comment]
Yes the issue seems to be that it is not pulling the .taskrc  tasks for display if you add a task it tells you the correct task number was created but it does not see the new task in the webui.

**Comment by norwat** [comment]
Confirming. After installing task-web with sudo no task is shown

**Comment by laomaiweng** [comment]
Ok I got it fixed, with taskwarrior-2.4.4 and taskwarrior-web-1.1.11.

First thing in `lib/taskwarrior-web/services/builder/base.rb:9`:

    -:query => TaskwarriorWeb::Config.version < '1.9.2' ? '_query' : 'export',
    +:query => 'export',

I don't know how this was supposed to work, the taskwarrior manpages don't mention `_query`. `export` seems to have been around for quite a while (as far back as 1.7.0), but the default format changed from CSV to JSON for 1.9.3 (this is consistent with what's found in `lib/taskwarrior-web/services/parser.rb`).

Next thing is to fix the broken JSON parsing in `lib/taskwarrior-web/services/parser/json.rb:7`:

    -json == '[No matches.]' ? [] : ::JSON.parse(json)
    +json == '[No matches.]' ? [] : ::JSON.parse(json.gsub(/\n/,','))

Otherwise as soon as there are several tasks, the resulting `json` string is just a newline-separated array-like thing instead of a well-formed comma-separated JSON array.

Hope that helps.

**Comment by theunraveler** [comment]
Submit a PR and I'll merge it.

**Comment by yangli1-stanford** [comment]
On Thu, May 28, 2015 at 7:13 AM, Quentin Minster <notifications@github.com>
wrote:

> json == '[No matches.]' ? [] : ::JSON.parse(json.gsub(/\n/,','))



It works, thank you.

**Comment by patschwork** [comment]
Also, it works/shows the tasks again. Many Thanks @laomaiweng :-)

But, editing (click/action is ignored) and annotating (shows an error page complaining about nothing to find with the GUID) doesn't work.
Also, if add a new task (works ;-) ), then taskserver is complaining with an error about a false priotity property. Beware of adding tasks via web in combination with use of taskserver (taskd)...
